code/modules/lavaland/lavaland_theme.dm

```diff
@@ -25,32 +25,24 @@
 	planet_icon_state = "planet_lava"
 
 /datum/lavaland_theme/lava/setup()
-	spawn_rivers(level_name_to_num(MINING))
+	var/datum/river_spawner/lava_spawner = new(level_name_to_num(MINING))
+	lava_spawner.generate()
 
 /datum/lavaland_theme/plasma
 	name = "plasma"
 	primary_turf_type = /turf/simulated/floor/lava/lava_land_surface/plasma
 	planet_icon_state = "planet_plasma"
 
 /datum/lavaland_theme/plasma/setup()
-	spawn_rivers(level_name_to_num(MINING), nodes = 2)
-	spawn_rivers(level_name_to_num(MINING), nodes = 2)
+	var/datum/river_spawner/spawner = new(level_name_to_num(MINING))
+	spawner.generate(nodes = 2)
+	spawner.generate(nodes = 2) // twice
```

Comment:
Why twice? If this is just a relic of the past, that's fine, I'm just curious if you know the motivation behind it